pydoctor/epydoc/markup/epytext.py

```diff
@@ -1352,69 +1352,101 @@ def to_stan(self, docstring_linker: DocstringLinker) -> Tag:
         if self._tree is None:
             self._stan = Tag('')
         else:
-            self._stan = self._to_stan(self._tree, docstring_linker)
+            self._stan = Tag('', children=node2stan.node2stan(self.to_node(), docstring_linker).children)
         return self._stan
-
-    def _to_stan(self,
-            tree: Union[Element, str],
-            linker: DocstringLinker,
-            seclevel: int = 0
-            ) -> Any:
+    
+    def to_node(self,) -> nodes.document:
+
+        if self._document is not None:
+            return self._document
+        
+        self._document = utils.new_document('epytext')
+        
+        if self._tree is not None:
+            children = list(self._to_node(self._tree))
+            assert len(children)==1
+            # The contents is encapsulated inside a section node. 
+            # Reparent the contents of the second level to the root level. 
+            self._document.children.extend(set_nodes_parent(children[0].children, self._document))
+        
+        return self._document
+    
+    def _to_node(self, tree: Union[Element, str]) -> Iterable[nodes.Node]:
```

Comment:
I see what you mean, but it looks like there are subtle differences. 

The following diff represents what I've tried, but it makes 33 test fails with assertions errors. I'm sure it's worth dedicating a lot of time trying to make this work since we have sufficient testing to stay with the current implementation in my opinion. 

```diff
diff --git a/pydoctor/epydoc/markup/epytext.py b/pydoctor/epydoc/markup/epytext.py
index 224219c0..a1434e7a 100644
--- a/pydoctor/epydoc/markup/epytext.py
+++ b/pydoctor/epydoc/markup/epytext.py
@@ -1371,16 +1371,15 @@ class ParsedEpytextDocstring(ParsedDocstring):
         
         return self._document
     
-    def _to_node(self, tree: Union[Element, str]) -> Iterable[nodes.Node]:
-        
-        if isinstance(tree, str):
-            yield set_node_attributes(nodes.Text(tree), document=self._document)
-            return
+    def _to_node(self, tree: Element) -> Iterable[nodes.Node]:
 
         # Process the children first.
         variables: List[nodes.Node] = []
-        for c in tree.children:
-            variables.extend(self._to_node(c))
+        for child in tree.children:
+            if isinstance(child, str):
+                variables.append(set_node_attributes(nodes.Text(tree), document=self._document))
+            else:
+                variables.extend(self._to_node(child))
 
         # Perform the approriate action for the DOM tree type.
         if tree.tag == 'para':
```

Reply:
sorry I just saw an issue in my code, I'll retry it..

Reply:
You forgot to substitute `child` for `tree` in this line:
```py
variables.append(set_node_attributes(nodes.Text(tree), document=self._document))
```

Reply:
Yes that's what I see. It's fixed now! Thanks